app/assets/javascripts/discourse/app/templates/components/sidebar/topics-section.hbs

```diff
@@ -8,26 +8,15 @@
   @headerAction={{this.composeTopic}}
   @headerActionTitle={{i18n "sidebar.sections.topics.header_action_title"}}>
 
-  <Sidebar::SectionLink
-    @linkName="everything"
-    @route={{this.everythingSectionLinkRoute}}
-    @query={{hash f=undefined}}
-    @title={{i18n "sidebar.sections.topics.links.everything.title"}}
-    @content={{i18n "sidebar.sections.topics.links.everything.content"}}
-    @current-when={{"discovery.latest discovery.new discovery.unread discovery.top"}}
-    @badgeText={{this.everythingSectionLinkBadgeText}} />
-
-  <Sidebar::SectionLink
-    @linkName="tracked"
-    @route="discovery.latest"
-    @query={{hash f="tracked"}}
-    @title={{i18n "sidebar.sections.topics.links.tracked.title"}}
-    @content={{i18n "sidebar.sections.topics.links.tracked.content"}} />
-
-  <Sidebar::SectionLink
-    @linkName="bookmarked"
-    @route="userActivity.bookmarks"
-    @model={{this.currentUser}}
-    @title={{i18n "sidebar.sections.topics.links.bookmarked.title"}}
-    @content={{i18n "sidebar.sections.topics.links.bookmarked.content"}} />
+  {{#each this.sectionLinks as |sectionLink|}}
+    <Sidebar::SectionLink
```

Comment:
I question why we don't just pass a sectionLink here instead of redefining all these properties?

Reply:
I wanted to keep the `Sidebar::SectionLink` component flexible instead of tying it directly to the `BaseSectionLink` class which is meant for the topics section at the moment. I'll keep this like this for now and see if a nicer pattern emerges once I start working on the categories section in the sidebar. I'll definitely keep your feedback in mind and will determine if this needs to be changed.